com.unity.render-pipelines.universal/Editor/ShaderPreprocessor.cs

```diff
@@ -169,33 +363,27 @@ bool StripUnusedFeatures(ShaderFeatures features, Shader shader, ShaderSnippetDa
                 return true;
             }
 
-            // strip main light shadows, cascade and screen variants
-            if (!IsFeatureEnabled(features, ShaderFeatures.MainLightShadows))
-            {
-                if (compilerData.shaderKeywordSet.IsEnabled(m_MainLightShadows))
-                    return true;
-
-                if (compilerData.shaderKeywordSet.IsEnabled(m_MainLightShadowsCascades))
-                    return true;
-
-                if (compilerData.shaderKeywordSet.IsEnabled(m_MainLightShadowsScreen))
-                    return true;
+            var stripOffVariants = UniversalRenderPipelineGlobalSettings.instance?.stripOffVariants ?? true;
+            var stripTool = new StripTool<ShaderFeatures>(features, shader, snippetData, compilerData.shaderKeywordSet, stripOffVariants);
 
-                if (snippetData.passType == PassType.ShadowCaster && !compilerData.shaderKeywordSet.IsEnabled(m_CastingPunctualLightShadow))
-                    return true;
-            }
+            // strip main light shadows, cascade and screen variants
+            // TODO: Strip disabled keyword once no light will re-use same variant
+            if (stripTool.StripMultiCompileKeepOffVariant(
+                m_MainLightShadows, ShaderFeatures.MainLightShadows,
+                m_MainLightShadowsCascades, ShaderFeatures.MainLightShadowsCascade,
+                m_MainLightShadowsScreen, ShaderFeatures.ScreenSpaceShadows))
+                return true;
 
-            bool isSoftShadow = compilerData.shaderKeywordSet.IsEnabled(m_SoftShadows);
-            if (!IsFeatureEnabled(features, ShaderFeatures.SoftShadows) && isSoftShadow)
+            // TODO: Strip off variants once we have global soft shadows option for forcing instead as support
```

Comment:
Even if soft shadows enabled, urp can still use disabled variant. This will be for next PR.